spec/features/beis_users_can_upload_level_b_activities_spec.rb

```diff
@@ -207,11 +158,137 @@
+
+      expect(rows).to eq([
+        "RODA ID",
+        "Parent RODA ID",
+        "Transparency identifier",
+        "Title",
+        "Description",
+        "Benefitting Countries",
+        "Partner organisation identifier",
+        "GDI",
+        "GCRF Strategic Area",
+        "GCRF Challenge Area",
+        "SDG 1", "SDG 2", "SDG 3",
+        "Newton Fund Pillar",
+        "Covid-19 related research",
+        "ODA Eligibility",
+        "Activity Status",
+        "Planned start date", "Planned end date",
+        "Actual start date", "Actual end date",
+        "Sector",
+        "Collaboration type (Bi/Multi Marker)",
+        "Aid type",
+        "Free Standing Technical Cooperation",
+        "Aims/Objectives",
+        "NF Partner Country PO",
+        "Comments"
+      ])
+    end
+
+    scenario "uploading a valid set of activities" do
+      old_count = Activity.count
+
+      within ".upload-form--non-ispf" do
+        attach_file_and_click_submit(filepath: "spec/fixtures/csv/valid_level_b_non_ispf_activities_upload.csv")
+      end
+
+      expect(Activity.count - old_count).to eq(2)
+
+      new_activities = [
+        Activity.find_by(title: "Programme - Award (round 5)"),
+        Activity.find_by(title: "Isolation and redesign of single-celled examples")
+      ]
+
+      visit organisation_activities_path(organisation)
+
+      within "//tbody" do
+        new_activities.each { |activity| expect(page).to have_content(activity.title) }
+      end
+
+      visit organisation_activity_comments_path(organisation, new_activities.first)
+
+      expect(page).to have_text("A comment")
+
+      visit organisation_activity_comments_path(organisation, new_activities.second)
+
+      expect(page).to have_text("Another comment")
+    end
+  end
+
+  context "ISPF ODA" do
+    scenario "downloading the CSV template" do
+      click_link t("action.activity.download.link", type: t("action.activity.type.ispf_oda"))
+
+      csv_data = page.body.delete_prefix("\uFEFF")
+      rows = CSV.parse(csv_data, headers: false).first
+
+      expect(rows).to eq([
+        "RODA ID",
+        "Parent RODA ID",
+        "Transparency identifier",
+        "Title",
+        "Description",
+        "Benefitting Countries",
+        "Partner organisation identifier",
+        "GDI",
+        "SDG 1",
+        "SDG 2",
+        "SDG 3",
+        "ODA Eligibility",
+        "Activity Status",
+        "Planned start date",
+        "Planned end date",
+        "Actual start date",
+        "Actual end date",
+        "Sector",
+        "Aid type",
+        "Aims/Objectives",
+        "ISPF theme",
+        "ISPF partner countries",
+        "Comments"
+      ])
+    end
+
+    scenario "uploading a valid set of activities" do
+      old_count = Activity.count
+
+      within ".upload-form--ispf-oda" do
+        attach_file_and_click_submit(filepath: "spec/fixtures/csv/valid_level_b_ispf_ooda_activities_upload.csv")
```

Comment:
I was confused by the filename: `OODA` is a specific, non-ISPF fund. Should this be `ispf_oda`?

Reply:
OOps